lib/nuclear_secrets.rb

```diff
@@ -51,15 +51,17 @@ def build_assertions(secrets, existing_keys)
             [
               build_secret_tuple(secrets, required_secrets, key),
             ],
-          )
+          ) if @settings[:raise_on_extra_secrets] == true
         end
-      end
+      end.compact
     end
 
     def check_assertions(secrets, assertions)
       secrets.to_a.zip(assertions).select do |pair|
-        result = pair.last.call(pair.first[1])
-        if !result
+        result = if pair.last.present?
```

Comment:
This is to support extra secrets right? Because `pair.last` wouldn't exist if it was an extra secret?

Reply:
Correct